content/blogs/investigating-wddm.md

```diff
@@ -0,0 +1,40 @@
+ CDD.dll is far more interesting on one hand it is an XDDM display driver as well. It also fires off two different IOCTLs which do the communication to WDDM. Indeed this driver is by all accounts the only route at which Dxgkrnl <-> win32k talk to each other in any meaningful capacity. Technically there’s an a bit of communication between watchdog, win32k and dxgkrnl to facilitate initialization and to fill in the interface to dispatch d3dkmt APIs down to DxgKrnl but that’s only during the start up of DxgKrnl itself. It’s also worth noting, when this happens the query in win32k to find a supported display adapter is always manually overwritten with cdd.dll. Once you start a WDDM driver you cannot run a XDDM at the same time! 
+	Alright well that was a lot of information but there’s some important things to understand here. CDD.dll first and foremost IS an XDDM display driver, even if it’s translating from the old world win32k to the new world WDDM stack it’s still stressing win32k out. This means for ReactOS simply to be truly compatible with WDDM and get vendor drivers to start up their display properly we need to make sure our XDDM stack is in great shape. This isn’t the only Vista+ feature that has this requirement, DWM itself (which is worth its own blog post) does a lot of things that even the current ReactOS win32k just isn’t capable of handling quite yet. But we're constantly improving.
+ (note reactos cdd on vista)
+### Compiling WDDM drivers with ReactOS
+One of the first components that needed to be understood was displib.lib a component shipped with the WDK that allows for compiling WDDM drivers. https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/dispmprt/nf-dispmprt-dxgkinitializedisplayonlydriver
+In order for you to compile a WDDM driver you need to link with a library that implements a function like the above to actually allow your driver to “start” DxgKrnl, as WDDM drivers don’t link against Dxgkrnl at all! Instead when you call an API like the above or it’s full WDDM twin you pass the information to DxgKrnl it needs to callback to your miniport’s initialization routine and this is what gives you the necessary interfaces to talk to DxgKrnl further. Win32k has nothing to do with your miniport driver starting up! This was relatively straightforward to make an alternative to, opening the door for ReactOS to import and compile WDDM drivers that even work on windows!
```

Comment:
your `DxgKrnl` capitalization is all over the place, I'd choose one and replace all instances with that.